staging_service/import_specifications/file_parser.py

```diff
@@ -24,6 +33,37 @@ class ErrorType(Enum):
     OTHER = 2
 
 
+@dataclass(frozen=True)
+class FileTypeResolution:
+    """
+    The result of resolving a file type given a file path.
+
+    Only one of type or unsupported_type may be specified.
+
+    type - the resolved file type if the type is one of the supported types.
+    unsupported_type - the file type if the type is not a supported type.
+    """
+    type: SupportedFileType = None
+    unsupported_type: str = None
+
+    def __post_init__(self):
+        if not (bool(self.type) ^ bool(self.unsupported_type)):  # xnor
+            raise ValueError("Exectly one of type or unsupported_type must be supplied")
+
+
+@dataclass(frozen=True)
+class PathDef:
+    """
+    A definition of a path to a file.
+
+    file - the true path to the file. This may contain implementation details that should be
+        hidden from the user.
+    userfile - the path to the file that should be displayed to the user.
+    """
+    file: Path
+    userfile: Path
```

Comment:
I'm not actually sure if I need the userfile at this point. I'll finish the parser and then switch the PathDef for a simple Path if it turns out I don't.